animation_nodes/nodes/mesh/find_shortest_path.py

```diff
@@ -0,0 +1,93 @@
+import bpy
+from bpy.props import *
+from ... data_structures import Mesh, LongList
+from ... base_types import AnimationNode, VectorizedSocket
+from ... algorithms.mesh_generation.find_shortest_path import getShortestPath, getShortestTree
+
+modeItems = [
+    ("PATH", "Path", "Find path from source vertex to other vertex", 0),
+    ("TREE", "Tree", "Find paths from source vertex(ies) to other vertices", 1)
+]
+
+pathTypeItems = [
+    ("MESH", "Mesh", "Output paths as line mesh(es)", 0),
+    ("SPLINE", "Spline", "Output paths as poly splines", 1),
+    ("STROKE", "Stroke", "Output paths as gp strokes", 2)
+]
+
+class FindShortestPathNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_FindShortestPathNode"
+    bl_label = "Find Shortest Path"
+    errorHandlingType = "EXCEPTION"
+
+    mode: EnumProperty(name = "Mode Type", default = "TREE",
+        items = modeItems, update = AnimationNode.refresh)
+
+    pathType: EnumProperty(name = "Path Type", default = "MESH",
+        items = pathTypeItems, update = AnimationNode.refresh)
+
+    joinMeshes: BoolProperty(name = "Join Meshes", default = True,
+        update = AnimationNode.refresh)
+
+    useSourceList: VectorizedSocket.newProperty()
+
+    def create(self):
+        self.newInput("Mesh", "Mesh", "mesh")
+        if self.mode == "PATH":
+            self.newInput("Integer", "Source", "source")
+            self.newInput("Integer", "Target", "target", value = 1)
+        else:
+            self.newInput(VectorizedSocket("Integer", "useSourceList",
+                    ("Source", "sources"), ("Sources", "sources")))
+
+        if self.mode == "PATH":
+            self.newOutput("Integer List", "Indices", "indices")
+        else:
+            if self.pathType == "MESH":
+                if not self.joinMeshes:
+                    self.newOutput("Mesh List", "Meshes", "outMeshes")
+                else:
+                    self.newOutput("Mesh", "Mesh", "outMesh")
+            elif self.pathType == "SPLINE":
+                self.newOutput("Spline List", "Splines", "outSplines")
+            elif self.pathType == "STROKE":
+                self.newOutput("GPStroke List", "Strokes", "outStrokes")
+
+    def draw(self, layout):
+        layout.prop(self, "mode", text = "")
+        layout.prop(self, "pathType", text = "")
```

Comment:
This shouldn't be drawn if the mode is PATH.

Reply:
Done.